### PR TITLE
DOC: Reorganize contents of stats User Guide section

### PR DESCRIPTION
**Pull request by melissawm**
#### Reference issue
Related to #16191

#### What does this implement/fix?
Reorganizes contents of the stats User Guide section to move content into dedicated pages and only list a ToC in the main landing page. This allows readers to have a high level overview of all contents before clicking.

Also adds a section dedicated to Hypothesis tests, empty so far. This can be populated later.

No content is changed in this PR, only the organization/ToC is affected.

cc @mdhaber

### CODE REVIEW

**Review by mdhaber — COMMENTED** [review]
The idea looks good, based on the diff, but I'll check more closely when it's done rendering. Thanks @melissawm!

One thing is that we might not need a separate "Hypothesis tests" section, if we're going to have separate one sample and two sample (or multiple sample) sections. They cover hypothesis tests, and links to the new examples could go in the respective sections.

**Comment by melissawm** [comment]
There's a couple of missing references I didn't catch locally but will fix now. Thanks @mdhaber - that sounds reasonable as well. Let me check we are linking all of the tests there just to make sure we are not missing any content.

**Comment by mdhaber** [comment]
What do you think about this order/nesting arrangement?

- Random Variables 
  - Maybe renamed to "Probability distributions" for consistency with the API reference?
  - Also, let's link to Discrete Statistical Distributions and Continuous Statistical Distributions from here, since they really aren't "tutorials" but mathematical reference information.
  - I also thinki "Building specific distributions" can be a subheading within the Random Variables section.
- Sample Statistics and Hypothesis Tests
  - Analyzing one sample
  - Analyzing multiple samples
  - Resampling and Monte Carlo Methods
- Universal Non-Uniform Random Number Sampling (this probably could be nested with probability distributions or combined with Quasi-Monte Carlo under a more general heading, but OK to leave it separate)
- Quasi-Monte Carlo

**Comment by melissawm** [comment]
Here's how the landing page for stats looks like right now: https://output.circle-artifacts.com/output/job/26164f43-7916-4131-b3a0-7790d416ca1c/artifacts/0/html/tutorial/stats.html

I have left a toctree depth of 2 because I think it gives visibility to other subsections, but now we can reorganize if we feel like there's irrelevant stuff exposed in the landing page. Also happy to keep experimenting 😄

File: doc/source/tutorial/stats/building_distributions.rst [hunk 1]
```diff
@@ -0,0 +1,172 @@
+Building specific distributions
```
Comment (mdhaber) [inline doc/source/tutorial/stats/building_distributions.rst:1]: Let's move this to the bottom of `probability_distributions.rst`.

File: doc/source/tutorial/stats/kernel_density_estimation.rst [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+    >>> plt.show()
+
+.. plot:: tutorial/stats/plots/kde_plot5.py
+   :align: center
+   :alt: "An X-Y plot showing a random scattering of points around a 2-D gaussian. The distribution has a semi-major axis at 45 degrees with a semi-minor axis about half as large. Each point in the plot is highlighted with the outer region in red, then yellow, then green, with the center in blue. "
+   :include-source: 0
+
+
+Multiscale Graph Correlation (MGC)
```
Comment (mdhaber) [inline doc/source/tutorial/stats/kernel_density_estimation.rst:180]: Let's move this to its own file listed below Kernel Density Estimation.

File: doc/source/tutorial/stats/probability_distributions.rst [hunk 3]
```diff
@@ -0,0 +1,447 @@
+Probability distributions
+-------------------------
+
+.. toctree::
+    :maxdepth: 1
+
+    continuous
+    discrete
+    building_distributions
```
Comment (mdhaber) [inline doc/source/tutorial/stats/probability_distributions.rst:9]: Per another comment, `building_distributions` would be removed from this list and the content moved to the bottom of this document.

Can we include links to the `continuous` and `discrete` pages after there is some context like below?

File: doc/source/tutorial/stats/probability_distributions.rst [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+
+
+There are two general distribution classes that have been implemented
+for encapsulating :ref:`continuous random variables
+<continuous-random-variables>` and :ref:`discrete random variables
+<discrete-random-variables>`. Over 80 continuous random variables
+(RVs) and 10 discrete random variables have been implemented using
+these classes. Besides this, new routines and distributions can be
+easily added by the end user. (If you create one, please contribute it.)
```
Comment (mdhaber) [inline doc/source/tutorial/stats/probability_distributions.rst:18]:
```suggestion
these classes. For mathematical reference information about individual
distributions, please see:

<links to continuous and discrete distribution reference pages>

Besides these, new distributions can be added by the end user.
```

**Review by mdhaber — APPROVED** [review]
Reviewed by roughly confirming that the old content is preserved somewhere and that the new structure appropriate for the existing content. We'll be updating a lot of these in future PRs, so I am not suggesting any changes to content here.

Just a few hierarchy suggestions, and then I think this is good to go.

File: doc/source/tutorial/stats/probability_distributions.rst (hunk 4 above)
Comment (melissawm) [reply doc/source/tutorial/stats/probability_distributions.rst]: I captured this change but also added a ToC later in the document highlighting the distributions. Let me know what you think.

File: doc/source/tutorial/stats/probability_distributions.rst (hunk 3 above)
Comment (melissawm) [reply doc/source/tutorial/stats/probability_distributions.rst]: I have done both: moved the ToC for continuous and discrete distributions down for previous context, and also added the text you suggested. We can leave both or simplify, it's up to you. Thanks!

**Comment by mdhaber** [comment]
Thanks @melissawm!

**Pull request merged**